### PR TITLE
home tasks 21.02.2019 

### PR DESCRIPTION
**Pull request by MateAcademy**

### CODE REVIEW

**Review by Daniluko — APPROVED** [review]
Why you copy Abstarct Pattern? You must understand this pattern and you can rework your example

File: src/main/java/com/mateacademy/patterny/abstractFactory/AuctionSiteProject.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.mateacademy.patterny.abstractFactory;
+
+import com.mateacademy.patterny.abstractFactory.website.WebsiteTeamFactory;
+
+public class AuctionSiteProject {
+    public static void main(String[] args) {
+        ProjectTeamFactory projectTeamFactory = new WebsiteTeamFactory();
```
Comment (SerhiiPasko) [inline src/main/java/com/mateacademy/patterny/abstractFactory/AuctionSiteProject.java:7]: You use concretic factory|class. It is hardcode. Read https://www.baeldung.com/java-abstract-factory-pattern

File: src/main/java/com/mateacademy/patterny/adapter/DatabaseRunner.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.mateacademy.patterny.adapter;
+
+import java.text.DateFormat;
+
+public class DatabaseRunner {
+    public static void main(String[] args) {
+        Database database = new AdapterJavaToDatabase();
```
Comment (SerhiiPasko) [inline src/main/java/com/mateacademy/patterny/adapter/DatabaseRunner.java:7]: You have "JavaApplication" and must use it as "Database". I don't see that

File: src/main/java/com/mateacademy/patterny/decorator/DeveloperDecorator.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.mateacademy.patterny.decorator;
+
+public class DeveloperDecorator implements Developer{
+
+    Developer developer;
```
Comment (SerhiiPasko) [inline src/main/java/com/mateacademy/patterny/decorator/DeveloperDecorator.java]: private

Comment (SerhiiPasko) [reply src/main/java/com/mateacademy/patterny/decorator/DeveloperDecorator.java]: DeveloperDecorator implements Developer - bad name

File: src/main/java/com/mateacademy/patterny/proxy/ProxyProject.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    private RealProject realProject;
+
+    public ProxyProject(String url) {
+        this.url = url;
+    }
+
+
+    @Override
+    public void run() {
+        if (realProject == null) {
```
Comment (SerhiiPasko) [inline src/main/java/com/mateacademy/patterny/proxy/ProxyProject.java:15]: And what beatifull add your proxy?

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
Read comments

File: src/main/java/com/mateacademy/patterny/abstractFactory/AuctionSiteProject.java (hunk 1 above)
Comment (MateAcademy) [reply src/main/java/com/mateacademy/patterny/abstractFactory/AuctionSiteProject.java:7]: I add new commit: commit 21.02.2019 - abstrack factory 2 where i fixed

File: src/main/java/com/mateacademy/patterny/decorator/DeveloperDecorator.java (hunk 3 above)
Comment (MateAcademy) [reply src/main/java/com/mateacademy/patterny/decorator/DeveloperDecorator.java]: I add  commit: commit 21.02.2019 private Developer

File: src/main/java/com/mateacademy/patterny/proxy/ProxyProject.java (hunk 4 above)
Comment (MateAcademy) [reply src/main/java/com/mateacademy/patterny/proxy/ProxyProject.java:15]: I showed the goal proxy: replacing another object and controlling access to it

Comment (MateAcademy) [reply src/main/java/com/mateacademy/patterny/proxy/ProxyProject.java:15]: we have some Prodject, we have implementation RealProject and we have implementation ProxyProject and we hahe a client project: ProjectRunner which  causes ProxyProject, which contains as an element RealProject  and inside yourself creates this real project

File: src/main/java/com/mateacademy/patterny/abstractFactory/AuctionSiteProject.java (hunk 1 above)
Comment (SerhiiPasko) [reply src/main/java/com/mateacademy/patterny/abstractFactory/AuctionSiteProject.java:7]: Ok. I see there same as before.

**Review by Daniluko — CHANGES_REQUESTED** [review]
Why you copy Abstract Pattern? You must understand this pattern and you can rework your example

File: src/main/java/com/mateacademy/patterny/adapter/DatabaseRunner.java (hunk 2 above)
Comment (SerhiiPasko) [reply src/main/java/com/mateacademy/patterny/adapter/DatabaseRunner.java:7]: I don't see real changes

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
Read comments

**Pull request merged**